torch/distributed/device_mesh.py

```diff
@@ -188,7 +188,7 @@ def create_flatten_mesh(
             # Check whether the mesh_dim_name for flattened mesh is valid.
             self.flatten_name_to_root_dims.setdefault(root_mesh, {})
             invalid_dim_names = chain(
-                *list(not_none(root_mesh.mesh_dim_names)),
+                list(not_none(root_mesh.mesh_dim_names)),
```

Comment:
No tests for this regression?

Reply:
you're saying that its pointless to flatten a single dim, right?
well, shouldn't we still add a test for how flatten behaves in this case?  if we want to throw an exception we can assertRaises..